### PR TITLE
Named sandbox secrets

### PR DESCRIPTION
**Pull request by edwardfoyle**
## Problem

Sandbox secret commands do not recognize the `--name` argument. This means that there is no way to manage secrets for named sandboxes from the CLI

**Issue number, if available:**
fixes: https://github.com/aws-amplify/amplify-backend/issues/1125
## Changes

Modified the `--name` arg to be global to all sandbox commands and make necessary plumbing refactors to use the name arg in the `sandbox secret` commands. I also included some type refactoring that wasn't strictly required, but makes the types across the arg objects more consistent

**Corresponding docs PR, if applicable:**
https://github.com/aws-amplify/docs/pull/7108

## Checklist

- [x] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [x] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [x] If this PR requires a docs update, I have linked to that docs PR above.
- [x] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: packages/cli/src/commands/sandbox/sandbox_command.ts [hunk 1]
```diff
@@ -184,7 +183,7 @@ export class SandboxCommand
             const projectNameRegex = /^[a-zA-Z0-9-]{1,15}$/;
             if (!argv.name.match(projectNameRegex)) {
               throw new Error(
-                `--name should match [a-zA-Z0-9-] and less than 15 characters.`
+                `--name should match [a-zA-Z0-9-] and be less than 15 characters.`
```
Comment (0618) [inline packages/cli/src/commands/sandbox/sandbox_command.ts:186]: 15 characters for kabab case is hard

File: packages/cli/src/commands/sandbox/sandbox-secret/sandbox_secret_set_command.test.ts [hunk 2]
```diff
@@ -71,6 +71,28 @@ void describe('sandbox secret set command', () => {
...
   });
 
+  void it('sets a secret in a named sandbox', async (contextual) => {
+    const mockSecretValue = contextual.mock.method(
+      AmplifyPrompter,
+      'secretValue',
+      () => Promise.resolve(testSecretValue)
+    );
+
+    await commandRunner.runCommand(`set ${testSecretName} --name anotherName`);
```
Comment (0618) [inline packages/cli/src/commands/sandbox/sandbox-secret/sandbox_secret_set_command.test.ts:81]: I thought we only accept kabab-case? Why is the test using camelCase?

**Review by 0618 — APPROVED** [review]

File: packages/cli/src/commands/sandbox/sandbox-secret/sandbox_secret_set_command.test.ts (hunk 2 above)
Comment (0618) [reply packages/cli/src/commands/sandbox/sandbox-secret/sandbox_secret_set_command.test.ts:81]: sounds like "camelKabab-case" 😂

File: packages/cli/src/commands/sandbox/sandbox_command.ts [hunk 3]
```diff
@@ -80,16 +78,18 @@ export class SandboxCommand
...
 
     // attaching event handlers
     const clientConfigLifecycleHandler = new ClientConfigLifecycleHandler(
       this.clientConfigGeneratorAdapter,
-      args['config-version'] as ClientConfigVersion,
-      args['config-out-dir'],
-      args['config-format']
+      args.configVersion as ClientConfigVersion,
+      args.configOutDir,
+      args.configFormat
```
Comment (rtpascual) [inline packages/cli/src/commands/sandbox/sandbox_command.ts:92]: Should we also do this for other commands as well for consistency?

**Review by rtpascual — APPROVED** [review]
LGTM, just one comment

**Review by Amplifiyer — APPROVED** [review]

**Pull request merged**